src/pages/[platform]/build-a-backend/add-aws-services/overriding-resources/index.mdx

```diff
@@ -57,47 +57,44 @@ The `backend` object exposes a `resources` property with objects for each of the
 For example, here is how you can access the Cognito user pool that is created by `defineAuth` and set a custom removal policy on the resource.
 
 ```ts title="amplify/backend.ts"
+import { RemovalPolicy } from 'aws-cdk-lib';
 import { defineBackend } from '@aws-amplify/backend';
 import { auth } from './auth/resource';
-import { UserPool } from 'aws-cdk-lib/aws-cognito';
-import { RemovalPolicy } from 'aws-cdk-lib';
 
 const backend = defineBackend({
   auth
 });
 
-const userPool = backend.auth.resources.userPool as UserPool;
+const userPool = backend.auth.resources.userPool;
 userPool.applyRemovalPolicy(RemovalPolicy.RETAIN_ON_UPDATE_OR_DELETE);
 ```
 
 Most L1 and L2 AWS CDK constructs that are used by the `define*` functions are accessible in this way.
 
 ## Example - Grant access permissions between resources
 
-Consider the case that we want to grant a function created by `defineFunction` access to call the Cognito user pool created by `defineAuth`. This can be accomplished with the following overrides.
+Consider the case that we want to grant a function created by `defineFunction` access to call the Cognito user pool created by `defineAuth`. For most cases it is recommended to use the [`access` property on `defineAuth`](/[platform]/build-a-backend/auth/grant-access-to-auth-resources/), however for permissions not exposed by this property, access can be accomplished with the following overrides.
 
 ```ts title="amplify/backend.ts"
 import { defineBackend } from '@aws-amplify/backend';
 import { auth } from './auth/resource';
 import { data } from './data/resource';
-import { demoFunction } from './functions/demo-function/resource';
-import { UserPool } from 'aws-cdk-lib/aws-cognito';
-import { Function } from 'aws-cdk-lib/aws-lambda';
+import { authAuditorFunction } from './functions/auth-auditor-function/resource';
 
 const backend = defineBackend({
   auth,
   data,
-  demoFunction
+  authAuditorFunction,
 });
 
-const userPool = backend.auth.resources.userPool as UserPool;
-const lambdaFunction = backend.demoFunction.resources.lambda as Function;
+const userPool = backend.auth.resources.userPool;
+const lambdaFunction = backend.authAuditorFunction.resources.lambda;
 
-// grant the lambdaFunction read access to users
-userPool.grant(lambdaFunction, 'cognito:GetUser', 'cognito:ListUsers');
+// grant the lambdaFunction access to list auth events for a particular user
+userPool.grant(lambdaFunction, 'cognito:AdminListUserAuthEvents');
 
 // pass the Lambda the UserPool ID so that the Lambda can use it to make SDK calls
-lambdaFunction.addEnvironment('USER_POOL_ID', userPool.userPoolId);
+backend.authAuditorFunction.addEnvironment('USER_POOL_ID', userPool.userPoolId);
 ```
 
 ## Example - Mutate synthesized CloudFormation
```

Comment:
Nit: not this PR change, but we are not mutating _synthesized_ cloudformation templates. We are modifying L1 constructs that get eventually synthesized to CFN template

Reply:
good callout, added a change to this heading